compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,216 @@ module Language.Bond.Codegen.Java.Class_java
+        javaFieldDefaultValueInitParamExpr BT_UInt8 (Just (DefaultInteger val)) = [lt|(byte)#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_UInt16 (Just (DefaultInteger val)) = [lt|(short)#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_UInt32 (Just (DefaultInteger val)) = [lt|#{val}|]
+        javaFieldDefaultValueInitParamExpr BT_UInt64 (Just (DefaultInteger val)) = [lt|#{val}L|]
+        javaFieldDefaultValueInitParamExpr BT_Float (Just (DefaultFloat val)) = [lt|#{val}F|]
+        javaFieldDefaultValueInitParamExpr BT_Double (Just (DefaultFloat val)) = [lt|#{val}D|]
+        javaFieldDefaultValueInitParamExpr BT_String (Just (DefaultString val)) = [lt|"#{val}"|]
+        javaFieldDefaultValueInitParamExpr BT_WString (Just (DefaultString val)) = [lt|"#{val}"|]
+        javaFieldDefaultValueInitParamExpr (BT_UserDefined e@Enum {..} _) (Just (DefaultEnum val)) = [lt|#{qualifiedDeclaredTypeName java e}.#{val}|]
+        javaFieldDefaultValueInitParamExpr t _ = error $ "invalid declaration type for fieldDefaultValueInitParamExpr: " ++ (show t)
+
+
+-- builds text for GenericBondTypeBuilder abstract class
+genericBondTypeBuilderDeclMakeText :: String -> [TypeParam] -> Text
+genericBondTypeBuilderDeclMakeText declName declParams = [lt|
+    public static abstract class GenericBondTypeBuilder {
+
+        private GenericBondTypeBuilder() {
+        }
+
+        public abstract <#{paramList}> com.microsoft.bond.StructBondType<#{declName}<#{paramList}>> makeGenericType(#{sepBy ", " methodArg declParams});
+    }
+|]
+    where
+        paramList = sepBy ", " paramName declParams
+        methodArg TypeParam {..} = [lt|com.microsoft.bond.BondType<#{paramName}> #{paramName}|]
+
+
+-- builds text for implementation of the GenericBondTypeBuilder.makeGenericType method
+makeGenericTypeMethodImplMakeText :: String -> [TypeParam] -> Text
+makeGenericTypeMethodImplMakeText declName declParams = [lt|
+            final <#{paramList}> com.microsoft.bond.StructBondType<#{declName}<#{paramList}>> makeGenericType(#{sepBy ", " methodArg declParams}) {
+                #{newlineSep 3 checkArgument declParams}
+                return (__StructBondTypeImpl<#{paramList}>) (com.microsoft.bond.StructBondType)this.getInitializedFromCache(#{paramList});
+            }
+|]
+    where
+        paramList = sepBy ", " paramName declParams
+        methodArg TypeParam {..} = [lt|com.microsoft.bond.BondType<#{paramName}> #{paramName}|]
+        checkArgument TypeParam {..} = [lt|com.microsoft.bond.helpers.ArgumentHelper.ensureNotNull(#{paramName}, "#{paramName}");|]
+
+
+-- builds text for anonymous implementation of the GenericBondTypeBuilder abstract class and assignment to the BOND_TYPE variable
+bondTypeStaticVariableDeclAsGenericBondTypeBuilder :: String -> [TypeParam] -> Text
+bondTypeStaticVariableDeclAsGenericBondTypeBuilder declName declParams = [lt|public static final GenericBondTypeBuilder BOND_TYPE = new GenericBondTypeBuilder() {
```

Comment:
Not sure what you mean by "casting in constructors", please clarify.

---
In reply to: [125371147](https://github.com/Microsoft/bond/pull/517#discussion_r125371147) [](ancestors = 125371147,125136658)